python/seldon_core/microservice.py

```diff
@@ -462,13 +473,39 @@ def grpc_prediction_server():
         except (NotImplementedError, AttributeError):
             pass
 
-        server.add_insecure_port(f"0.0.0.0:{grpc_port}")
-
+        server.add_insecure_port(bind_address)
         server.start()
+        _wait_forever(server)
+
+    @contextlib.contextmanager
+    def _reserve_grpc_port():
+        """Find and reserve a port for all subprocesses to use."""
+        sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        if sock.getsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT) != 1:
+            raise RuntimeError("Failed to set SO_REUSEPORT.")
+        sock.bind(('', grpc_port))
+        try:
+            yield sock.getsockname()[1]
+        finally:
+            sock.close()
 
-        logger.info("GRPC microservice Running on port %i", grpc_port)
-        while True:
-            time.sleep(1000)
+    def grpc_prediction_server():
+        with _reserve_grpc_port() as bind_port:
+            bind_address = '0.0.0.0:{}'.format(bind_port)
+            logger.info(f"GRPC Server Binding to '%s' {bind_address} with {args.workers} processes")
+            sys.stdout.flush()
+            workers = []
+            for _ in range(args.workers):
+                # NOTE: It is imperative that the worker subprocesses be forked before
+                # any gRPC servers start up. See
+                # https://github.com/grpc/grpc/issues/16001 for more details.
+                worker = multiprocessing.Process(
+                    target=_run_grpc_server, args=(bind_address,))
+                worker.start()
+                workers.append(worker)
+            for worker in workers:
+                worker.join()
```

Comment:
It appears this doens't work with seldon 1.10: 
```  
File "/usr/local/lib/python3.7/dist-packages/seldon_core/microservice.py", line 506, in grpc_prediction_server
    worker.start()
  File "/usr/lib/python3.7/multiprocessing/process.py", line 110, in start
    'daemonic processes are not allowed to have children'
```

Reply:
Can we just start without daemon True at https://github.com/SeldonIO/seldon-core/blob/4781cc5c5bd415ea3c6562f9918209ca66d64241/python/seldon_core/microservice.py#L62

Reply:
yep, looks like it